src/api/app/controllers/source_controller.rb

```diff
@@ -371,7 +371,9 @@ def update_file
       end
     end
 
-    Package.verify_file!(@pack, params[:filename], request.raw_post)
+    readable = proc { request.raw_post }
+    readable.define_singleton_method(:read) { call }
+    Package.verify_file!(@pack, params[:filename], readable)
```

Comment:
It's probably me being stupid but why do you add all this code instead of passing the request object into `Package.verify_file!`? The `request` object already supports all the things you want, no?